### DISCUSSION TITLE
Different results between segmentation plot and inference results

### DISCUSSION THREAD

**Discussion by amitli1** [pyannote/pyannote-audio#1328] [Development]
I'm trying to run `speaker segmentation` on test  file (sample rate = 16000, file duration = 24.951 seconds) and get strange results.

I run the following code:
```
from pyannote.audio              import Model
from pyannote.audio              import Inference
SAMPLE_WAV = "file.wav"
segment_model = Model.from_pretrained("pyannote/segmentation")
inference = Inference(segment_model, duration=5.0, step=2.5)
sg_output = inference(SAMPLE_WAV)

```
`sg_output`:

Results:
![image](https://user-images.githubusercontent.com/128060257/232692462-ef75d226-b4f5-4da7-8d41-2ae068c273e0.png)

I tried to get the relevant results of window number 6 and got different results:
```
import plotly.express       as px 
import pandas               as pd
import numpy                as np

# --- choose window
i              = 5 # window #6
current_window = sg_output.data[i]

# --- get relvant times and probailties
start_time     = i*2.5
end_time       = start_time + 5 
arr_time       = np.arange(start = start_time, 
                           stop  = end_time, 
                           step  = segment_model.introspection.frames.step)

prob_len       = len(current_window[:, 0])

# --- ? the arr_time is smaller than prob_len, why ?
arr_time       = arr_time[0: prob_len]

# --- create dataframe
arr_speakers   = ["Speaker A"] * len(arr_time) + ["Speaker B"] * len(arr_time)  + ["Speaker C"] * len(arr_time)
arr_times      = np.concatenate((arr_time, arr_time, arr_time))
arr_prob       = np.concatenate((current_window[:, 0], current_window[:, 1], current_window[:, 2]))

df             = pd.DataFrame()
df["Time"]     = arr_times
df["Prob"]     = arr_prob
df["Speaker"]  = arr_speakers
fig            = px.line(df, x ="Time", y="Prob", color="Speaker")

fig.update_layout(
    title       = dict(text = "Speaker Segmentation", font=dict(size=24)),
    xaxis_title = "Time", 
    yaxis_title = "Probability"
)

fig.show()
```

![image](https://user-images.githubusercontent.com/128060257/232692757-e18501c3-d15e-418f-80d5-7f6e83f58557.png)

- Why there is a difference between those 2 results? 
- Does taking the value of `sg_output.data` give the appropriate window?
- Where is the mistake

**Comment by hbredin** [comment]
Window number 6 (index 5) is not the one your circled in black.
It is the one on the third line centered on t=15s.

**Reply by hbredin** [reply]
![image](https://user-images.githubusercontent.com/1080837/232696695-b2722d00-6944-456f-9d57-ae6fd6da2fff.png)

**Reply by amitli1** [reply]
As you wrote, I got confused with the index.
Still, I'm confused why there is a difference between the length of `arr_time = 297` and the window length (`len(current_window[:, 0]) = 293`) ?
